src/test/java/util/xcmailr/util/SendRequest.java

```diff
@@ -2,145 +2,137 @@
 
-            String output;
-            while ((output = br.readLine()) != null)
+            if (emailObject.size() > 0)
             {
-                sb.append(output);
+                return emailObject.get(0).getAsJsonObject();
             }
-            postRequest.releaseConnection();
+            return null;
         }
         catch (IOException e)
         {
-            try
-            {
-                br.close();
-            }
-            catch (IOException e1)
-            {
-                e1.printStackTrace();
-            }
-            e.printStackTrace();
             throw new RuntimeException(e);
         }
+    }
+
+    public static boolean emailExists(String email)
+    {
+        return getTempEmail(email) != null;
+    }
+
+    public static boolean emailExpired(String email)
+    {
+        return getTempEmail(email).get("expired").getAsBoolean();
+    }
+
+    public static void deleteTempEmail(String tempEmail)
+    {
+        String id = getTempEmail(tempEmail).get("id").getAsString();
+        try
+        {
+            final HttpPost postRequest = new HttpPost(XcMailrApi.getConfiguration().url() + "/mail/delete/" + id);
+            httpClient.execute(postRequest);
 
+            postRequest.releaseConnection();
+        }
+        catch (IOException e)
+        {
+            throw new RuntimeException(e);
+        }
     }
 
-    private static String getIdOfTempEmail(String tempEmail)
+    private static StringBuilder readResponse(InputStream inputStream)
     {
-        HttpResponse response = null;
         StringBuilder sb = new StringBuilder();
         BufferedReader br = null;
         try
         {
-            HttpGet getRequest = new HttpGet("https://xcmailr.xceptance.de/mail/getmails");
+            br = new BufferedReader(new InputStreamReader(inputStream, "UTF-8"));
 
-            response = httpClient.execute(getRequest);
-            br = new BufferedReader(new InputStreamReader(response.getEntity().getContent(), "UTF-8"));
-
-            String output;
-            while ((output = br.readLine()) != null)
+            sb = new StringBuilder();
+            for (String output = br.readLine(); output != null; output = br.readLine())
             {
                 sb.append(output);
             }
-            getRequest.releaseConnection();
-            JSONArray emails = (JSONArray) new JSONParser(JSONParser.MODE_JSON_SIMPLE).parse(sb.toString());
-            Long id = (Long) ((JSONObject) ((JSONArray) JsonPath.read(emails, "$[?(@.fullAddress=='" + tempEmail + "')]")).get(0)).get("id");
-            return id.toString();
         }
-        catch (IOException | ParseException e)
+        catch (IOException e)
+        {
+            throw new RuntimeException("couldn't read the response from server", e);
+        }
+        finally
         {
             try
             {
                 br.close();
```

Comment:
If we cannot create the buffered reader, this throws a NPE. Please only close it if br is not null.